File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        for (int i=0; i<n; i++) {
            String map1 = Integer.toString(arr1[i], 2);
            String map2 = Integer.toString(arr2[i], 2);
            map1 = "0".repeat(n-map1.length()) + map1;
            map2 = "0".repeat(n-map2.length()) + map2;
            
            String line = "";
            for (int j=0; j<n; j++) {
                line += Integer.parseInt(""+map1.charAt(j)) | Integer.parseInt(""+map2.charAt(j));
            }
            line = line.replaceAll("1", "#");
            line = line.replaceAll("0", " ");
            answer[i] = line;
            
        }
        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    var answer = [];
    
    for (let i=0; i<n; i++) {
        let c = (arr1[i] | arr2[i]).toString(2).padStart(n, '0');
        c = c.replaceAll('1', '#');
        c = c.replaceAll('0', " ");
        answer.push(c);
    }
    
    return answer;
}